src/execution/mod.rs

```diff
@@ -111,42 +112,42 @@ impl<'b> RuntimeInstance<'b> {
         wasm.move_start_to(inst.code_expr);
 
         loop {
-            match wasm.read_u8().unwrap_validated() {
+            match wasm.read_instruction().unwrap_validated() {
```

Comment:
could this be instead:

```rust
match wasm[self.pc..] {
```

Reply:
good question, will go for this and talk more with @george-cosma that has helped me a lot discussing different ways, hope to get to a solution that is perfect for everyone, both from an aesthetic point of view, but also from a performance point of view

Reply:
Done